src/services/util/helpers.ts

```diff
@@ -0,0 +1,24 @@
+/**
+ * Takes an object containing possible configuration arguments and their associated values,
+ * and then filters out any configuration argument whose value is undefined. The result is an array of
+ * individual configuration argument objects, each consisting of a property and its corresponding value.
+ *
+ * @param {Object} obj - An object containing possible configuration arguments and values.
+ * @returns {Array} An array of individual configuration argument objects with valid values.
+ */
+export function createConfigArgsList(obj:{[key: string]: any}): {[key: string]: any}[]{
+    console.log(obj)
+    let argsList: {[key: string]: any}[] = [];
+    for(const key of Object.keys(obj)){
+        console.log(key, obj[key])
+        if(obj[key] != undefined){
+            const arg: {[key: string]: any} = {};
+            arg[key] = obj[key];
+            argsList.push(
+               arg
+            )
+        }
+    }
```

Comment:
to avoid nesting you can filter out `undefined` values first and then do the assigning in-place using map

```
const argsList = Object.keys(obj)
        .filter(key => obj[key] !== undefined)
        .map(key => ({ [key]: obj[key] })); 
```

this will remove your logs though. but let me know if this works.

Reply:
Oh wow, that's super useful! It works --> d763fe972e618236bd64763544e9f3fdc8ad0043  I didn't need those logs anyway actually